src/GitHub/Endpoints/PullRequests.hs

```diff
@@ -60,6 +67,45 @@ pullRequestsForR user repo opts = pagedQuery
     ["repos", toPathPart user, toPathPart repo, "pulls"]
     (prModToQueryString opts)
 
+-- | Obtain the diff of a pull request
+-- See <https://developer.github.com/v3/pulls/#get-a-single-pull-request>
+pullRequestDiff' :: Maybe Auth -> Name Owner -> Name Repo -> Id PullRequest -> IO (Either Error ByteString)
+pullRequestDiff' auth user repo prid =
+    executeRequestMaybe auth $ pullRequestDiffR user repo prid
+
+-- | Obtain the diff of a pull request
+-- See <https://developer.github.com/v3/pulls/#get-a-single-pull-request>
+pullRequestDiff :: Name Owner -> Name Repo -> Id PullRequest -> IO (Either Error ByteString)
+pullRequestDiff = pullRequestDiff' Nothing
+
+-- | Query a single pull request to obtain the diff
+-- See <https://developer.github.com/v3/pulls/#get-a-single-pull-request>
+pullRequestDiffR :: Name Owner -> Name Repo -> Id PullRequest -> Request k ByteString
+pullRequestDiffR user repo prid =
+    RawHeaderQuery
```

Comment:
is this header still required?

Reply:
Ok. I see, we need this to get `ByteString`. I'd prefer a concrete sum type for `Diff`, `Patch` and sha from https://developer.github.com/v3/media/#commits-commit-comparison-and-pull-requests